In [1]:
import pandas as pd
import numpy as np
import datetime
import warnings
import matplotlib.pyplot as plt
import seaborn as sns

warnings.filterwarnings('ignore')

In [45]:
!pip install detect-delimiter==0.1.1
  

In [46]:
from detect_delimiter import detect

In [47]:
detect('../lab-t-tests-p-values/files_for_lab/machine.txt')# I didn't manage to get the data
#from the .txt format the right way, so I changed the file format using excel into .csv

'-'

In [88]:
data = pd.read_csv('../lab-t-tests-p-values/files_for_lab/machine.csv',sep="\;")
#r"C:\Users\hydro\a.txt", sep='\t'
data.columns = ['New_Machine','Old_Machine']

In [89]:
data

New_Machine  Old_Machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [90]:
data.columns

Index(['New_Machine', 'Old_Machine'], dtype='object')

In [100]:
from scipy.stats import ttest_ind, norm

In [101]:
ttest_ind(data['New_Machine'], data['Old_Machine'])

Ttest_indResult(statistic=-3.3972307061176026, pvalue=0.0032111425007745158)

### results: 
The Null Hypothesis, that the new and the old machines would produce with the same speed, cannot be rejected. Because the p-value is 0.003 < 0.05.
So it cannot be stated, that the new machine is faster than th old one.

In [106]:
data_university = pd.read_csv('../lab-t-tests-p-values/files_for_lab/student_gpa.csv',sep="\;")
#r"C:\Users\hydro\a.txt", sep='\t'
data_university.columns = ['Sophomores','Juniors']

In [117]:
data_university

Sophomores   Juniors
0         3.04  2.560000
1         1.71  2.770000
2         3.30  2.700000
3         2.88  3.000000
4         2.11  2.980000
5         2.60  3.470000
6         2.92  3.260000
7         3.60  3.200000
8         2.28  3.190000
9         2.82  2.650000
10        3.03  3.000000
11        3.13  3.390000
12        2.86  2.580000
13        3.49  2.980769
14        3.11  2.980769
15        2.13  2.980769
16        3.27  2.980769

In [116]:
data_university=data_university.fillna(data_university.mean())

In [121]:
data_university['Sophomores'].mean()
#data_university['Juniors'].mean()

2.8400000000000003

In [118]:
ttest_ind(data_university['Sophomores'], data_university['Juniors'])

Ttest_indResult(statistic=-0.9925381999259013, pvalue=0.3283813595175433)

### results:

The Null Hypothesis, that the Juniors and Sophomores of the university would have the same gpa mean, can be rejected with a p-value of 0.33 > 0.05.
The Sophomores having a lower gpa mean.